### DISCUSSION TITLE
How to manually parse a document stored as an Object

### DISCUSSION THREAD

**Discussion by jon-m-bottoms** [ravendb/ravendb#16539] [Q&A] [answered]
For debugging purposes I would like take a document, as raw json from Raven.Studio, and deserialize into my proper c# object, without having to store or retrieve in from a DocumentStore

I've configured Newtonsoft using the DefaultRavenContractResolver and this works for the top level fields/objects.

`MyObject myObject = JsonConvert.DeserializeObject<MyObject>(jsonContent,
                    new JsonSerializerSettings
                    {
                        ContractResolver = new DefaultRavenContractResolver(new DocumentConventions().Serialization)
                    });`

However, MyObject contains a collection of Events where are stored as an object type (c#)
` "Events": [
        {
            "Item": {
                "$type": "MyClass.Events.Event1, MyClass.Events",
		"Field1": "aaa",
               ...}
           ...
        },
        {
            "Item": {
                "$type": "MyClass.Events.Event1, MyClass.Events",
		"Field2": "bbb",
                ...}
          ...
        },`

These are being deserialized as a Newtonsoft JObject instead of the underlying $type. I'm assuming I'm just missing a setting in the serializer, but it's not obvious to me what that is. 

Any thoughts?

Thanks

**Comment by ppekrol** [comment, marked as answer]
Hi @jon-m-bottoms 

IIRC, during deserialization you need to set TypeNameHandling to Auto. Check here: https://www.newtonsoft.com/json/help/html/serializetypenamehandling.htm

**Reply by jon-m-bottoms** [reply]
Yup, that did the trick. Thanks so much!
